lib/sprockets/less/v4/importer.rb

```diff
@@ -0,0 +1,108 @@
+          filename = check_path_before_process(context, base_path)
+          base_path = (filename.is_a?(Pathname) ? filename : Pathname.new(filename))
+          super(context, path, base_path)
+        end
+
+        def check_path_before_process(context, path, a = nil)
+          if path.to_s.start_with?('file://')
+            path = Pathname.new(path.to_s.gsub(/\?type\=(.*)/, "?type=text/#{syntax}"))  # @TODO : investigate why sometimes file:/// URLS are ending in ?type=text instead of ?type=text/scss
+            asset = context.environment.load(path) # because resolve now returns file://
+            asset.filename
+          else
+            path
+          end
+        end
+
+        def stat_of_pathname(context, pathname, _path)
+          filename = check_path_before_process(context, pathname)
+          context.environment.stat(filename)
+        end
+
+        # @TODO find better alternative than scanning file:// string for mime type
+        def content_type_of_path(context, path)
+          pathname = context.resolve(path)
+          content_type = pathname.nil? ? nil : pathname.to_s.scan(/\?type\=(.*)/).flatten.compact.first unless pathname.nil?
+          attributes = {}
+          [content_type, attributes]
+        end
+
+        def get_engines_from_attributes(_context, _attributes)
+           []
+         end
+
+
+        def get_context_transformers(context, content_type, path)
+          available_transformers =  context.environment.transformers[content_type]
+          additional_transformers = available_transformers.key?(syntax_mime_type) ? available_transformers[syntax_mime_type] : []
+          additional_transformers.is_a?(Array) ? additional_transformers : [additional_transformers]
+          css_transformers = available_transformers.key?('text/css') ? available_transformers['text/css'] : []
+          css_transformers = css_transformers.is_a?(Array) ? css_transformers : [css_transformers]
+          additional_transformers = additional_transformers.concat(css_transformers)
+          additional_transformers
+        end
+
+        def filter_all_processors(processors)
+          processors.delete_if do |processor|
+            filtered_processor_classes.include?(processor) || filtered_processor_classes.any? do |filtered_processor|
+              !processor.is_a?(Proc) && ((processor.class != Class && processor.class < filtered_processor) || (processor.class == Class && processor < filtered_processor))
+            end
+          end
+        end
+
+        def call_processor_input(processor, context, input, processors)
+          if processor.respond_to?(:processors)
+            processor.processors = filter_all_processors(processor.processors)
+          end
+          super(processor, context, input, processors)
+        end
+
+        def filtered_processor_classes
+          classes = super
+          classes << Sprockets::YUICompressor if defined?(Sprockets::YUICompressor)
+          classes
```

Comment:
I fancy to use `super.tap { |classes| classes << Sprockets::YUICompressor if defined?(Sprockets::YUICompressor) }` in those situations. Not a required change though.